### ISSUE TITLE
Problem mapping nested byte array to denormalized property

### ISSUE THREAD

**Issue by filiphr** [mapstruct/mapstruct#978]
```java
@Mapper
public interface SimpleMapper {

    @Mapping( target = "props.byteArray", source = "byteArray" )
    SourceRoot toSourceRoot(TargetObject targetObject);
}
```

Will create a mapping for the byte array that looks like:

```java
public class SimpleMapperImpl implements SimpleMapper {
    @Override
    public SourceRoot toSourceRoot(TargetObject targetObject) {
        if ( targetObject == null ) {
            return null;
        }

        SourceRoot sourceRoot = new SourceRoot();

        SourceProps props = new SourceProps();
        sourceRoot.setProps( props );

        if ( targetObject.getByteArray() != null ) {
            byte[] props.byteArray = targetObject.getByteArray();
            props.setByteArray( Arrays.copyOf( props.byteArray, props.byteArray.length ) );
        }

        return sourceRoot;
    }
}
```

The `targetProperyName` is directly used from the mapping itself.

**Comment by sjaakd** [comment]
ouch.. that should be simple to fix..  I prefer taking the property name from the target type nowadays..

**Comment by filiphr** [comment]
I am working on it. Even if we use that target type we will have a problem when the mapping is reverse. I'll make a PR later tonight or tomorrow.

**Comment by filiphr** [comment]
Closed via PR #979
